Source/model/draw/Pattern.swift

```diff
@@ -1,12 +1,16 @@
 open class Pattern: Fill {
 
+    public let viewBox: Rect
     public let content: Node
     public let bounds: Rect
     public let userSpace: Bool
+    public let position: Transform
 
-    public init(content: Node, bounds: Rect, userSpace: Bool = false) {
+    public init(content: Node, bounds: Rect, viewBox: Rect, userSpace: Bool = false, position: Transform) {
```

Comment:
You need to keep in mind that this is a breaking change, because `Pattern` is a part of public API. New attributes should always have default value. I'm also not quite sure that `viewBox` should have separate attribute, need to think about it.